src/compiler/utilities.ts

```diff
@@ -82,17 +82,21 @@ namespace ts {
         return node.end - node.pos;
     }
 
-    export function arrayIsEqualTo<T>(arr1: T[], arr2: T[], comparer?: (a: T, b: T) => boolean): boolean {
-        if (!arr1 || !arr2) {
-            return arr1 === arr2;
+    export function arrayIsEqualTo<T>(array1: T[], array2: T[], equaler?: (a: T, b: T) => boolean,
+        sortBeforeComparison?: boolean, comparer?: (a: T, b: T) => number): boolean {
+        if (!array1 || !array2) {
+            return array1 === array2;
         }
 
-        if (arr1.length !== arr2.length) {
+        if (array1.length !== array2.length) {
             return false;
         }
 
-        for (let i = 0; i < arr1.length; ++i) {
-            let equals = comparer ? comparer(arr1[i], arr2[i]) : arr1[i] === arr2[i];
+        let newArray1 = sortBeforeComparison ? array1.slice().sort(comparer) : array1;
+        let newArray2 = sortBeforeComparison ? array2.slice().sort(comparer) : array2;
+
+        for (let i = 0; i < array1.length; ++i) {
+            let equals = equaler ? equaler(newArray1[i], newArray2[i]) : newArray1[i] === newArray2[i];

```

Comment:
if you don't want to sort before comparing, you don't always have a comparer to use though.